core/src/streamer.rs

```diff
@@ -205,4 +286,36 @@ mod test {
         t_receiver.join().expect("join");
         t_responder.join().expect("join");
     }
+
+    #[test]
+    pub fn streamer_test_deserialize_packets_in_blob() {
+        let meta = Meta::default();
+        let serialized_meta_size = serialized_size(&meta).unwrap() as usize;
+        let serialized_packet_size = serialized_meta_size + PACKET_DATA_SIZE;
+        let num_packets = 10;
+        let mut rng = rand::thread_rng();
+        let packets: Vec<_> = (0..num_packets)
+            .map(|_| {
+                let mut packet = Packet::default();
+                for i in 0..packet.meta.addr.len() {
```

Comment:
Could this be reduced to:

```rust
packet.meta.addr = packet.meta.addr.map(|x| rng.gen_range(1, std::u16::MAX));
packet.data = packet.data(|x| rng.gen_range(1, std::u8::MAX));
```

Reply:
Unfortunately you can't collect directly into an array: https://stackoverflow.com/questions/26757355/how-do-i-collect-into-an-array. You can do something like this, but I think you would first have to collect the iterators into a collection like a vector, and then copy that vector into the fixed size array. It seems more efficient to just do in-place writes like we have here in these loops

Reply:
I wrote that wrong. I meant:

```rust
packet.meta.addr = packet.meta.addr.into_iter().map(|x| rng.gen_range(1, std::u16::MAX)).collect();
packet.data = packet.data.into_iter().map(|x| rng.gen_range(1, std::u8::MAX)).collect();
```

Reply:
you can't collect an iterator into an array (both packet.meta.addr and packet.data are arrays) so you would have to first collect into a separate collection, and then copy them into the packet.meta.addr and packet.data arrays, which is less efficient than just the loops.